ShoppingApp/Cart.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ShoppingApp
+{
+    public class Cart
+    {
+        private Dictionary<Product, int> _addedProducts = new Dictionary<Product, int>();
+        private DiscountConfig _discountConfig;
+
+        public Cart(DiscountConfig discountConfig)
+        {
+            _discountConfig = discountConfig;
+        }
+
+        public double GetTotal()
+        {
+            double total = 0;
+            foreach(var product in _addedProducts.Keys)
+            {
+                total += product.Price * _addedProducts[product];
+            }
+            return total;
+        }
+        public double GetDiscountedTotal()
+        {
+            double total = 0;
+            foreach (var product in _addedProducts.Keys)
+            {
+                total += (product.Price * (1 - _discountConfig.GetDiscount(product.Category) / 100)) * _addedProducts[product];
```

Comment:
what will happen if we get default discount zero??

Reply:
then Discounted Price will be equal to Total Price.